milli/src/update/index_documents/transform.rs

```diff
@@ -92,7 +96,7 @@ impl Transform<'_, '_> {
         // We extract the primary key from the first document in
         // the batch if it hasn't already been defined in the index
         let first = documents.peek().and_then(|r| r.as_ref().ok());
-        let alternative_name = first.and_then(|doc| doc.keys().find(|k| k.contains(DEFAULT_PRIMARY_KEY_NAME)).cloned());
+        let alternative_name = first.and_then(|doc| doc.keys().find(|f| is_primary_key(f)).cloned());
```

Comment:
```suggestion
        let alternative_name = first.and_then(|doc| doc.keys().find(is_primary_key).cloned());
```

Reply:
not possible, f is a &&String, and rustc doesn't like it.